### ISSUE TITLE
Update JCIO Agency Report

### ISSUE THREAD

**Issue by Franceswog** [jac-uk/admin#2290]
## Background
Paul O'Brien raised this issue on 24/01/24, while working on [Exercise JAC00178 Chairman of the Competition Appeal Tribunal](https://admin.judicialappointments.digital/exercise/jFohU9YooUvW2qsEixIz/reports/agency#jcio), when he discovered the JCIO tab under Agency reports was blank. This runs the risk that professional checks may be incomplete. Additionally, the JCIO Agency export does not appear to correctly identify which candidates require a judicial check.

**Guidance** 
According to the JAC Selection Policy Guide, Judicial Conduct and Investigations Office (JCIO) checks are character checks existing fee-paid and salaried judicial office holders, i.e.:
1 Magistrate, for example, where a candidate is a magistrate, a JCIO check should be made even if the role applied for is a lay role
2 Existing judicial office holder (fee paid) 
3 Existing judicial office holder (salaried) 
4 Tax chamber - financial role (or where an issue of concern is identified) for fee-paid judicial office holder
5 Tax chamber - financial role (or where an issue of concern is identified) for salaried judicial office holder

The JAC does not consider lay magistrates (sometimes referred to as Justice of the Peace) as being a judicial office holder. However, as JCIO is the professional body responsible for them, you must identify all lay magistrates and ensure you consult with JCIO for those candidates.

## User Story
As an Admin user, I want the platform to trigger/generate a visible JCIO report identifying all candidates who have declared an appointment as a judicial office holder so that, when I click on the JCIO tab, I can view the details of candidates who require a JCIO check.

<img width="1647" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeHI1QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--621087087bf31870ada78e9472e89b6dd3f0bcf3/image.png" alt="image.png" />

## Feature(s) Description
Following the rollout of the updates to the application form, it seems the only logical way of identifying all candidates who have declared an appointment as a judicial office holder, and to ensure they are included in JCIO checks, is to:

- [ ] Set the trigger for inclusion on the Agency Reports JCIO tab to search for candidates who, within the 'post qualification experience' section of the application, have selected 'The carrying out of judicial functions in any court or tribunal' as a relevant law related task
- [ ] Ensure that the Export data download on this tab corresponds with the information displayed onscreen.

## User Testing Steps
- Check JCIO Agency tab is populated with correct candidates
- Check JCIO Agency export is populated as above

## Acceptance Criteria
 - [ ] **Scenario: JCIO tab report generation**
Given the JCIO tab is clicked, when the JCIO report is generated, then the report should be visible and populated with candidate data based on them having ticked 'The carrying out of judicial functions in any court or tribunal' as a relevant law related task
  
- [ ] **Scenario: JCIO checks export data**
Given the Export data link is clicked, the data downloaded corresponds with what is displayed onscreen

**Ticket Champion**
Paul O'Brien

**Comment by nickaddy** [comment]
@MattHowley Is this a flaw in the platform or just admin user error? I don't quite understand why we would want to discount or include judicial experience at exercise level?

**Comment by nickaddy** [comment]
@tomlovesgithub Can you pls confirm what the trigger on the platform is for candidates to appear under the JCIO Agency reports tab? If we can understand that, then we should be able to get to the bottom of what has happened here with Paul's exercise.

**Comment by tomlovesgithub** [comment]
The issue here comes fro the fact that fee-paid posts have been declared under the equality and diversity information. 
The full report shows the job titles declared by candidates, however gathers the Judicial Office data from the question within the judicial experience form on their application. 
The question is asked as “Are you a fee-paid or salaried judge?“, and if judicial experience is not configured for this exercise the candidates would not have been asked this question.

In short, it appears this is currently working as expected, however this might not be sufficient for its use, if so, let's discuss how this could be improved.

**Comment by MattHowley** [comment]
Hi @tomlovesgithub @nickaddy @Franceswog - think it definitely worth touching base please, not least given whether previous judicial experience is a requirement or not shouldn't be the trigger for JCIO checks; JCIO checks should be carried out on all judicial office holders, irrespective of whether PJE is a requisite for appointment. Having recently reconfigured the application form it is timely to look at what/when an application triggers a need for a JCIO check.

**Comment by Franceswog** [comment]
@MattHowley @nickaddy @tomlovesgithub   
I agree on the blanket JCIO checks for duty holder. Also, that the application form be revised to include question(s) that require candidates to disclose - previous history including Judicial appointments, professional conduct, legal proceeding, etc. 

The root cause of the failure to trigger a JCIO report is yet to be determined. 

Consider including information on the application form to educate candidates (on the importance of the JCIO checks etc.) and that checks will be conducted on all applicants. This will encourage transparency.

**Comment by tomlovesgithub** [comment]
DEVELOPER NOTE

`line 140 digital-platform/generateAgencyReport.js`

`````
      jcioOffice: helpers.toYesNo(application.feePaidOrSalariedJudge) || null,
      // ^^JCIO = said yes to fee paid....
      jcioPosts: application.experience ? application.experience.map(e => e.jobTitle).join(', ') : null,
      // any experience list job title
````

**Comment by MattHowley** [comment]
Hi @tomlovesgithub and co. Following the rollout of the updates to the application form, it seems the only logical way of identifying all candidates who have declared an appointment as a judicial office holder, and to ensure they are included in JCIO checks, is to search for candidates who, within the 'post qualification experience' section of the application, have selected 'The carrying out of judicial functions in any court or tribunal' as a relevant law related task

**Comment by NormaJAC** [comment]
I have tweaked the user stories and added additional information for more context to this ticket.

**Comment by nickaddy** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Development-5ea838cd2aec471eb6d14139/issues/jac-uk/admin/2290?planning-poker) with Zenhub @drieJAC @HalcyonJAC @tomlovesgithub @warrensearle
